crabs/detection_tracking/models.py

```diff
@@ -0,0 +1,157 @@
+import os
+import pickle
+import tempfile
+import time
+from datetime import datetime
+
+import mlflow
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torchvision
+from torch.utils.data import DataLoader
+from torch.utils.tensorboard import SummaryWriter
+from torchvision.models.detection.faster_rcnn import FastRCNNPredictor
+
+
+def create_faster_rcnn(num_classes: int, coco_model: bool = True) -> nn.Module:
+    """Create faster rcnn model for the training
+
+    Parameters
+    ----------
+    num_classes : int
+        number of classes to train
+    coco_moder : bool
+        either the dataset format is coco format or not
+
+    Returns
+    -------
+    model : nn.Module
+        the created model in this case Faster RCNN model
+    """
+
+    # Load Faster RCNN pre-trained model
+    model = torchvision.models.detection.fasterrcnn_resnet50_fpn_v2(
+        weights=torchvision.models.detection.FasterRCNN_ResNet50_FPN_V2_Weights.DEFAULT
+    )
+    if coco_model:  # Return the COCO pretrained model for COCO classes.
+        return model
+
+    # Get the number of input features
+    in_features = model.roi_heads.box_predictor.cls_score.in_features
+    # define a new head for the detector with required number of classes
+    model.roi_heads.box_predictor = FastRCNNPredictor(in_features, num_classes)
+
+    return model
+
+
+def train_faster_rcnn(
+    config,
+    model: nn.Module,
+    train_dataloader: DataLoader,
+    optimizer: optim.Optimizer,
+) -> nn.Module:
+    """Train faster rcnn model
+
+    Parameters
+    ----------
+    config
+        config including hyperparameters
+    model: nn.Module
+        the created model for the training
+    train_dataloader: DataLoader
+        dataloader instance for the given dataset which contains images and labels
+    optimizer: optim.Optimizer
+        optimizer instance for the given model
+
+    Returns
+    -------
+    model : nn.Module
+        the trained model
+    """
+
+    # select device (whether GPU or CPU)
+    device = (
+        torch.device("cuda")
+        if torch.cuda.is_available()
+        else torch.device("cpu")
+    )
+
+    # setup tensorboard stuff
+    layout = {
```

Comment:
nice, love being able to babysit the training with visualisation! 

I was wondering where is the MLFlow implementation? is it in a separate PR? (my understanding was that we were using MLflow to log so far, is that correct?)